### PR TITLE
Project: calendar improvements

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
This includes all the calendar improvements and the external calendars integration.

## :squid: Permissions
🚫 NO!

### CODE REVIEW

**Comment by juanlaube** [comment]
Some very important things for the review that are relevant for the rebranding release:
- there’s a new realm migration to add the new external calendar entitites
- there’s the new sync for those calendar entities which won’t work because backend just rolled out the endpoints on a different url than I my priorities where on other things so this is still broken, I will comment the lines that trigger this sync thing before to avoid any possible problem (it’s ok anyways, but we will have a lot of 404s)

File: Toggl.Droid/Startup/TogglApplication.cs [hunk 1]
```diff
@@ -127,5 +128,10 @@ public override void OnTrimMemory(TrimMemory level)
                     break;
             }
         }
+
+        private async void fetchExternalCalendarsDataIfNeeded()
+        {
+            await AndroidDependencyContainer.Instance.InteractorFactory.SyncExternalCalendars().Execute();
```
Comment (daividssilverio) [inline Toggl.Droid/Startup/TogglApplication.cs:134]: Why not simply comment out this? or 🔥

File: Toggl.iOS/Startup/AppDelegate.Background.cs [hunk 2]
```diff
@@ -103,5 +106,10 @@ private void detectTimerWidgetInstallStateChanged()
                 analyticsService.TimerWidgetInstallStateChange.Track(true);
             }
         }
+
+        private async void fetchExternalCalendarsDataIfNeeded()
+        {
+            await IosDependencyContainer.Instance.InteractorFactory.SyncExternalCalendars().Execute();
```
Comment (daividssilverio) [inline Toggl.iOS/Startup/AppDelegate.Background.cs:112]: Same question, why not comment it out? or 🔥.

**Review by daividssilverio — COMMENTED** [review]
My only concern is about the added models on Realm due to our recent crashes on iOS.
If we could test the migration on our test tracks... it would be really nice.

Considering that the visual changes are OK & minor feature changes in the calendar are also OK. This looks OK.

File: Toggl.Droid/Startup/TogglApplication.cs (hunk 1 above)
Comment (juanlaube) [reply Toggl.Droid/Startup/TogglApplication.cs:134]: @daividssilverio it was easier to just return earlier from the interactor 🤷‍♂️

**Review by dshr — APPROVED** [review]
Looks good, and I think the added Realm entities shouldn't impact much, since we're not populating the DB with them, and we make sure that doesn't happen with tests.

**Review by daividssilverio — APPROVED** [review]
I couldn't test the migration with the test builds.
But it worked alright when trying from rider on develop then this branch.
Let's make sure to test this throughly with the google play / test flight builds.

**Pull request merged**